### PR TITLE
Allow parameters for todo.sh default action

### PR DESCRIPTION
**Pull request by chrysle**
**Before submitting a pull request,** please make sure the following is done:

- [x] Fork [the repository](https://github.com/todotxt/todo.txt-cli) and create your branch from `master`.
- [x] If you've added code that should be tested, add tests!
- [x] Ensure the test suite passes.
- [x] Format your code with [ShellCheck](https://www.shellcheck.net/).
- [x] Include a human-readable description of what the pull request is trying to accomplish.
- [ ] Steps for the reviewer(s) on how they can manually QA the changes.
- [ ] Have a `fixes #XX` reference to the issue that this pull request fixes.

This finishes leftover work from #159 
That PR can be closed then.

I added a test, a docstring and also a note for `TODOTXT_DEFAULT_ACTION` to the config file.

### CODE REVIEW

**Comment by inkarkat** [comment]
Unfortunately, you're written the test to match the implementation, not to actually verify the specified behavior; and that actually was broken (good that we insisted on having tests). I've corrected both in my commit. It still doesn't handle default parameters that have spaces or other special characters; I'll add that later.

**Comment by chrysle** [comment]
Thanks for your help! What special parameters for example do you mean that should be tested?

**Comment by chrysle** [comment]
@inkarkat I would like to relieve you of implementing the tests if ou give me some hints.

File: tests/t0002-actions.sh [hunk 1]
```diff
@@ -37,4 +44,41 @@ test_expect_success 'custom action (env variable)' '
     test_cmp expect output && rm -rf myactions
 '
 
+test_expect_success 'custom action (default action)' '
+    mkdir .todo.actions.d
+    cp foo2 .todo.actions.d/
```
Comment (chrysle) [inline tests/t0002-actions.sh]: Maybe use the logical `&&` operator here too?

File: tests/t0002-actions.sh [hunk 2]
```diff
@@ -16,6 +16,13 @@ echo "TODO: foo"
 EOF
 chmod +x foo
 
+cat > foo2 << 'EOF'
+shift
+IFS=-
```
Comment (chrysle) [inline tests/t0002-actions.sh]: Why did you put dashes between the output?

File: todo.sh [hunk 3]
```diff
@@ -727,7 +727,13 @@ if [ -n "$OVR_TODOTXT_FINAL_FILTER" ] ; then
   TODOTXT_FINAL_FILTER="$OVR_TODOTXT_FINAL_FILTER"
 fi
 
-ACTION=${1:-$TODOTXT_DEFAULT_ACTION}
+isDefaultAction=
+if [ -n "$1" ]; then
+    ACTION=$1
+else
+    ACTION=$TODOTXT_DEFAULT_ACTION
+    isDefaultAction=t
```
Comment (chrysle) [inline todo.sh:787]: Maybe use `true` here to improve readability?

**Review by chrysle — COMMENTED** [review]
I have left some comments.

File: tests/t0002-actions.sh (hunk 1 above)
Comment (inkarkat) [reply tests/t0002-actions.sh]: Yes, makes sense.

File: tests/t0002-actions.sh (hunk 2 above)
Comment (inkarkat) [reply tests/t0002-actions.sh]: To be sure that the individual arguments are passed separately, and no wrong splitting on contained spaces occurs. I've added a comment to document this.

File: todo.sh (hunk 3 above)
Comment (inkarkat) [reply todo.sh:787]: The `test-lib.sh` already uses the short `t` for truth, and it's a common idiom.

**Comment by inkarkat** [comment]
Thanks for the review @chrysle; the original implementation was woefully incomplete and wrong, but only writing the tests showed what looked quite alright to a casual review. From my point of view this is now done, and we just have to wait for the merge of the huge backlog of open issues...

**Comment by chrysle** [comment]
Did you notice the test for macOS 10.15 succeeded? I wonder why it was always cancelled before.

**Review by inkarkat — APPROVED** [review]

**Pull request merged**